impeller/display_list/aiks_dl_vertices_unittests.cc

```diff
@@ -0,0 +1,356 @@
+      image, DlTileMode::kClamp, DlTileMode::kClamp, DlImageSampling::kLinear,
+      &matrix));
+
+  std::vector<SkPoint> positions = {SkPoint::Make(0, 0),
+                                    SkPoint::Make(size.width, 0),
+                                    SkPoint::Make(0, size.height)};
+  auto vertices =
+      MakeVertices(DlVertexMode::kTriangleStrip, positions, {0, 1, 2}, {}, {});
+
+  builder.DrawVertices(vertices, DlBlendMode::kSrcOver, paint);
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+// Regression test for https://github.com/flutter/flutter/issues/145707
+TEST_P(AiksTest, VerticesGeometryColorUVPositionData) {
+  DisplayListBuilder builder;
+  DlPaint paint;
+  auto image =
+      DlImageImpeller::Make(CreateTextureForFixture("table_mountain_nx.png"));
+  auto size = image->impeller_texture()->GetSize();
+
+  paint.setColorSource(std::make_shared<DlImageColorSource>(
+      image, DlTileMode::kClamp, DlTileMode::kClamp));
+
+  std::vector<SkPoint> positions = {
+      SkPoint::Make(0, 0),           SkPoint::Make(size.width, 0),
+      SkPoint::Make(0, size.height), SkPoint::Make(size.width, 0),
+      SkPoint::Make(0, 0),           SkPoint::Make(size.width, size.height),
+  };
+  std::vector<DlColor> colors = {
+      DlColor::kRed().withAlpha(128),   DlColor::kBlue().withAlpha(128),
+      DlColor::kGreen().withAlpha(128), DlColor::kRed().withAlpha(128),
+      DlColor::kBlue().withAlpha(128),  DlColor::kGreen().withAlpha(128),
+  };
+
+  auto vertices =
+      MakeVertices(DlVertexMode::kTriangles, positions, {}, {}, colors);
+
+  builder.DrawVertices(vertices, DlBlendMode::kDstOver, paint);
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, VerticesGeometryColorUVPositionDataAdvancedBlend) {
+  DisplayListBuilder builder;
+  DlPaint paint;
+  auto image =
+      DlImageImpeller::Make(CreateTextureForFixture("table_mountain_nx.png"));
+  auto size = image->impeller_texture()->GetSize();
+
+  paint.setColorSource(std::make_shared<DlImageColorSource>(
+      image, DlTileMode::kClamp, DlTileMode::kClamp));
+
+  std::vector<SkPoint> positions = {
+      SkPoint::Make(0, 0),           SkPoint::Make(size.width, 0),
+      SkPoint::Make(0, size.height), SkPoint::Make(size.width, 0),
+      SkPoint::Make(0, 0),           SkPoint::Make(size.width, size.height),
+  };
+  std::vector<DlColor> colors = {
+      DlColor::kRed().withAlpha(128),   DlColor::kBlue().withAlpha(128),
+      DlColor::kGreen().withAlpha(128), DlColor::kRed().withAlpha(128),
+      DlColor::kBlue().withAlpha(128),  DlColor::kGreen().withAlpha(128),
+  };
+
+  auto vertices =
+      MakeVertices(DlVertexMode::kTriangles, positions, {}, {}, colors);
+
+  builder.DrawVertices(vertices, DlBlendMode::kColorBurn, paint);
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+// Draw a hexagon using triangle fan
+TEST_P(AiksTest, CanConvertTriangleFanToTriangles) {
+  constexpr Scalar hexagon_radius = 125;
```

Comment:
No action required: What is the radius of a hexagon?  The distance between corners and the center?  Must be since there is a different `center_to_flat` which is probable the distance to the midpoint between corners.